File: 加密资料/DES-对称加密算法.py
import binascii

from pyDes import des, CBC, PAD_PKCS5


def des_encrypt(key, text, iv):
    k = des(key, CBC, iv, pad=None, padmode=PAD_PKCS5)
    en = k.encrypt(text, padmode=PAD_PKCS5)
    # print(en)
    return binascii.b2a_hex(en)


def des_decrypt(key, text, iv):
    k = des(key, CBC, iv, pad=None, padmode=PAD_PKCS5)
    de = k.decrypt(binascii.a2b_hex(text), padmode=PAD_PKCS5)
    return de


if __name__ == '__main__':
    secret_key = '12345678'  # 密钥
    text = 'hello world'  # 加密对象
    iv = secret_key  # 偏移量
    secret_str = des_encrypt(secret_key, text, iv)
    print('加密字符串：', secret_str)
    clear_str = des_decrypt(secret_key, secret_str, iv)
    print('解密字符串：', clear_str)

File: 加密资料/DES-对称加密算法.js
var cryptoJS =  require('crypto-js')

function encrypt_des() {
    var key = cryptoJS.enc.Utf8.parse(key)
    var iv = cryptoJS.enc.Utf8.parse(iv)
    var data = cryptoJS.enc.Utf8.parse(text)
    var enc_dta = cryptoJS.DES.encrypt(data, key, {
        iv: iv,
        mode: cryptoJS.mode.CBC,
        padding: cryptoJS.pad.Pkcs7
    })
    return enc_dta.toString()
}

function decrypt_des() {
    const key = cryptoJS.enc.Utf8.parse(key);
    const iv = cryptoJS.enc.Utf8.parse(iv);
    const data = enc_da;
    dec_data = cryptoJS.DES.decrypt(data, key, {
        iv: iv,
        mode: cryptoJS.mode.CBC,
        padding: cryptoJS.pad.Pkcs7
    })
    return dec_data.toString(cryptoJS.enc.Utf8)
}

var text = 'python'  // 明文
var key = '6f726c64f2c2057' // key
var iv = '0123456789ABCDEF' // 偏移值   初始向量

enc_da = encrypt_des()
console.log(decrypt_des());














